src/shared/Microsoft.Git.CredentialManager/HostProviderRegistry.cs

```diff
@@ -87,7 +88,7 @@ public void Register(IHostProvider hostProvider, HostProviderPriority priority)
             providers.Add(hostProvider);
         }
 
-        public Task<IHostProvider> GetProviderAsync(InputArguments input)
+        public async Task<IHostProvider> GetProviderAsync(InputArguments input)
```

Comment:
Nope! The reason is because the first commit didn't introduce any `await` keywords in the method body; adding the `async` keyword at that point would be redundant, and generate a build warning.

Adding `async` to the signature actually makes the compiler go off and generate the async state machine in IL. Since there is no `await` in here, we never made use of that generated magic (hence the would-be warning). I only add the `async` modifier once I make use of an async call that needs to be `await`ed in the subsequent commit(s).